dataset_api.md

```diff
@@ -0,0 +1,155 @@
+Incorporate the changes proposed from Martynas, with the exception of graphs(), which would now return a dictionary of graph names (URIRef or BNode) to Graph objects (as the graph's identifier would be removed).
+
+```
+add add_named_graph(uri: IdentifiedNode, graph: Graph) method
+add has_named_graph(uri: IdentifiedNode) method
+add remove_named_graph(uri: IdentifiedNode) method
+add replace_named_graph(uri: IdentifiedNode, graph: Graph)) method
+add graphs() method as an alias for contexts()
+add default_graph property as an alias for default_context
+add get_named_graph as an alias for get_graph
+deprecate graph(graph) method
+deprecate remove_graph(graph) method
+deprecate contexts() method
+Using IdentifiedNode as a super-interface for URIRef and BNode (since both are allowed as graph names in RDF 1.1).
+```
+
+Make the following enhancements to the triples, quads, and subject/predicate/object APIs.
+
+Major changes:
+P1. Remove `default_union` attribute and make the Dataset inclusive.
+P2. Remove the Default Graph URI ("urn:x-rdflib:default").
+P3. Remove Graph class's "identifier" attribute to align with the W3C spec, impacting Dataset methods which use the Graph class.
+P4. Make the graphs() method of Dataset return a dictionary of named graph names to Graph objects.
+Enhancements:
+P5. Support passing of iterables of Terms to triples, quads, and related methods, similar to the triples_choices method.
+P6. Default the triples method to iterate with `(None, None, None)`
+
+With all of the above changes, including those changes proposed by Martynas, here are some examples:
+
+```python
+from rdflib import Dataset, Graph, URIRef, Literal
+from rdflib.namespace import RDFS
+
+# ============================================
+# Adding Data to the Dataset
+# ============================================
+
+# Initialize the dataset
+d = Dataset()
+
+# Add a single triple to the Default Graph, and a single triple to a Named Graph
+g1 = Graph()
+g1.add(
+    (
+        URIRef("http://example.com/subject-a"),
+        URIRef("http://example.com/predicate-a"),
+        Literal("Triple A")
+    )
+)
+d.add_graph(g1)
```

Comment:
This method name is confusing imho. I first thaught it was adding a new named graph into the dataset with an auto-generated blank node for graph name. I would find cleaner to have a `Dataset.defaut_graph` accessor and then do things like `dataset.default_graph += g1` to merge data into the default graph and maybe allow to write `dataset.default_graph = g1` to replace the default graph.

Reply:
add_graph was an existing method but I agree that this is confusing, especially if add_named_graph is introduced and identifier is removed from Graph.

I think just having `Dataset.defaut_graph` is cleaner too - there's a method there already, renamed from `default_context`.

**If others see a benefit in retaining add_graph please comment.**